src/schema/posts.ts

```diff
@@ -740,6 +753,38 @@ export const resolvers: IResolvers<any, Context> = {
       }
       return { _: true };
     },
+    deleteSharedPost: async (
+      _,
+      { id }: { id: string },
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      await ctx.con.transaction(async (manager) => {
+        const repo = manager.getRepository(SharePost);
+        const post = await repo.findOneBy({ id });
+
+        if (!post?.sharedPostId) return;
+
+        await ensureSourcePermissions(
+          ctx,
+          post.sourceId,
+          SourcePermissions.PostDelete,
+        );
+
+        if (post.authorId !== ctx.userId) {
+          const sourceId = post.sourceId;
+          const memberRepo = manager.getRepository(SourceMember);
+          const [member, loggedUser] = await Promise.all([
+            memberRepo.findOneByOrFail({ sourceId, userId: post.authorId }),
+            memberRepo.findOneByOrFail({ sourceId, userId: ctx.userId }),
+          ]);
+          hasGreaterAccessCheck(loggedUser, member);
+        }
```

Comment:
Should this not all be inside the ensureSourcePermissions?

Maybe actually say 

if ctx.user is NOT the post owner, then evaluate if he is a mod or admin.

Reply:
I'm not sure we should. Some checks can vary from implementation to implementation like on the removal of members, we didn't have to check if the passed parameter is the logged-in user. Might make the function bloated with numerous conditions.

> if ctx.user is NOT the post owner, then evaluate if he is a mod or admin.

I think this is exactly the condition in line 773. Then the 780 evaluates if the user is a mod or admin and checks which has greater access privilege.

Reply:
But this particular section only evaluates the `postdelete` anyway, so not sure what it has to do with removal of members.

I just don't see where we actually have the `PostDelete` function?
That whole section seems to be not doing anything really.

Reply:
> But this particular section only evaluates the postdelete anyway, so not sure what it has to do with removal of members.

What I mean it has a similar process where it checks the privilege and their only difference is checking for the author which I think would add more complexity to the function if we pass another object which will only be used once.

> I just don't see where we actually have the PostDelete function?

The default value returned on the function is `true`, we have removed the `PostDelete` section since it is also now available for members. We will be using that enum value in the frontend to display the option of deleting the post. It could be not used here, but it will be utilized by the frontend.

Reply:
Let's discuss this on Gather rather, think we are meaning the same thing but not aligned on the implementation.